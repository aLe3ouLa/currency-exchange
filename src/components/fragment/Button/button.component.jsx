import React from "react";
import { ButtonContainer } from "./button.styles";

const CustomButton = ({ isDisabled, onClick, children }) => {
  return (
    <ButtonContainer disabled={isDisabled} onClick={onClick}>
      {children}
    </ButtonContainer>
  );
};

export default CustomButton;
