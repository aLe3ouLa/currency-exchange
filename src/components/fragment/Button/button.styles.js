import styled from "styled-components";

export const ButtonContainer = styled.button`
  background-color: ${({ theme }) => theme.color.primary};
  line-height: 1.5rem;
  color: ${({ theme }) => theme.color.white};
  font-weight: 500;
  font-size: 1.2rem;
  letter-spacing: 0.025em;
  border: none;
  width: 80%;
  padding: ${({ theme }) => theme.spacing.xs};
  box-shadow: ${({ theme }) => theme.color.shadow};
  border-radius: 2rem;
  outline: none;
  position: absolute;
  bottom: 20px;
  left: 50%;
  transform: translateX(-50%);

  transition: all 0.2s ease 0s;
  :hover {
    background-color: ${({ theme }) => theme.color.primaryDark};
  }
  &:disabled {
    opacity: 0.2;
    border-width: 0px;
    border-style: initial;
    border-color: initial;
    border-image: initial;
    cursor: auto;
  }
`;
