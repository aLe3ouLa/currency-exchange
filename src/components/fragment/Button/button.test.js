import React from "react";
import ReactDOM from "react-dom";
import Button from "./button.component";
import { cleanup, render, fireEvent } from "@testing-library/react";

import { ThemeProvider } from "styled-components";
import { theme, Defaults } from "../../../styles";

describe("Button tests", () => {
  afterEach(cleanup);

  const props = {
    isDisabled: false,
    onClick: jest.fn(),
    children: "Click me",
  };

  it("renders the Button component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Button {...props} />
        </React.Fragment>
      </ThemeProvider>,
      div
    );
  });

  it("has a valid snapshot ", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Button {...props} />
        </React.Fragment>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("fires click successfully", () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Button {...props} />
        </React.Fragment>
      </ThemeProvider>
    );
    fireEvent.click(getByText(/Click me/i));
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });

  it("disables button successully", () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Button {...props} isDisabled={true} />
        </React.Fragment>
      </ThemeProvider>
    );

    expect(getByText(/Click me/i)).toBeDisabled();
  });
});
