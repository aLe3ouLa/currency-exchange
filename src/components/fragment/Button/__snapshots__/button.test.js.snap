// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests has a valid snapshot  1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm FtxYm"
  >
    Click me
  </button>
</DocumentFragment>
`;
